mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -50,6 +50,9 @@
 
 import os, copy, re, pathlib
 import typing as T
+# T.cast is not handled by flake8 to detect quoted annotation use
+# see https://github.com/PyCQA/pyflakes/pull/632
+import typing
```

Comment:
Would it be possible to ignore the errors resulting from this instead until the linked PR is merged? I don't think that it is worth it to have two `import typing` statements just to make flake8 happy.

Reply:
It *could* be ignored, although then if the T.cast changed flake8 would not say anything.

Reply:
I was more thinking about something [like this](https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors).

Reply:
That's what I meant too. If the T.cast stopped using the noqa import, flake8 would not notice and say "hey, Interpreter is unused".

Reply:
Even if you explicitly specify the one error we want to get rid of with `# noqa: E<xyz>` ?

Reply:
for the record, pylint has the same issue: https://github.com/PyCQA/pylint/issues/3846. using `T` was a compromise between those of us who wanted the typing stuff in a namespace, and those who wanted it in not a namespace. But maybe the right solution is move to PEP585 annotations as forward annoations, then you could write:
```python
my_dict: `'dict[str, list[int]'`, instead of `typing.Dict[str, typing.List[int]]`, yes it means using strings, but at least we can vastly reduce the need for the typing module. Heck, we 3.9 we get a literal union syntax, `int | str` instead of `Union[int, str]`, which is pretty nice

Reply:
flake8 currently thinks Interpreter is unused. You can use noqa to say "no, darn it, always pretend it is used". But then if you actually do stop using it, the error is still specified to be gotten rid of.

Currently:

```python
import typing as T

if T.TYPE_CHECKING:
    from ..interpreter import Interpreter # used, but flake8 thinks it is unused and warns you

T.cast('Interpreter', self) # flake8 does not know this is the typing module and doesn't interpret type annotations
```

Proposed:
```python
import typing as T
import typing

if T.TYPE_CHECKING:
    from ..interpreter import Interpreter # used, and flake8 agrees

typing.cast('Interpreter', self) # flake8 sees a type annotation here
```

You want:

```python
from blah import BetterInterpreter
import typing as T

if T.TYPE_CHECKING:
    from ..interpreter import Interpreter # noqa: F401

T.cast('BetterInterpreter', self)

class Thing(BetterInterpreter):
    ...
```

Due to misc rewrites, BetterInterpreter is used both as an annotation and at runtime, and Interpreter isn't used at all. But the noqa is still there...

tl;dr I have a mild dislike of ignoring errors.

Reply:
@eli-schwartz I get your point, but I don't think that the risk is that great here, especially since all the code inside the `if T.TYPE_CHECKING:` is never actually run and just there for mypy.

@dcbaker I would wait with the `dict[str, list[str]]` syntax until we can use `from __future__ import annotations` in 3.7, then we don't have to use strings anymore.

Reply:
I don't think the risk of using `typing` is great either, but I concede and have went and used noqa.

And yes, I'm eager to use `__future__.annotations` too. :D

Sadly it has been delayed as the default mode (which it was supposed to be in 3.10) until at least 3.11, and it's not clear when it may become the default at all... :(